### PR TITLE
Typescriptify challenge services

### PR DESCRIPTION
**Pull request by asleepysheepy**
# Description of pull request

* type scripts `challenge-service`, challenge-start-service` and `challenge-cache`

# Issues Resolved

Part 6 of X associated with #74

### CODE REVIEW

File: src/challenges/challenge-cache.ts [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+import War from './models/war'
+import { Client, Collection, GuildChannel, Message, Snowflake, User } from 'discord.js'
+
+declare const client: Client
+
+enum FieldOptions {
+  X_STATUS = 'Cross-Server Display',
+  AUTO_STATUS = 'Automatic Summaries',
+  PREFIX = 'Prefix',
+  ANNOUNCE = 'Announcements Channel',
```
Comment (lisushka) [inline src/challenges/challenge-cache.ts:17]: Only the first two of these are speficially relevant to challenges?  Were they handled here before?

Comment (asleepysheepy) [reply src/challenges/challenge-cache.ts:17]: https://github.com/aigeroni/Winnie_Bot/blob/main/src/challenges/clist.js#L156

Comment (lisushka) [reply src/challenges/challenge-cache.ts:17]: Look who doesn't remember their own code!  😖

**Review by lisushka — APPROVED** [review]
Ship it!  👍

**Pull request merged**